### PR TITLE
encryption scope sas

### PR DESCRIPTION
**Pull request by microzchang**
# Pull Request Checklist

Please leverage this checklist as a reminder to address commonly occurring feedback when submitting a pull request to make sure your PR can be reviewed quickly:

See the detailed list in the [contributing guide](https://github.com/Azure/azure-sdk-for-cpp/blob/main/CONTRIBUTING.md#pull-requests).

- [ ] [C++ Guidelines](https://azure.github.io/azure-sdk/cpp_introduction.html)
- [ ] Doxygen docs
- [ ] Unit tests
- [ ] No unwanted commits/changes
- [ ] Descriptive title/description
  - [ ] PR is single purpose
  - [ ] Related issue listed
- [ ] Comments in source
- [ ] No typos
- [ ] Update changelog
- [ ] Not work-in-progress
- [ ] External references or docs updated
- [ ] Self review of PR done
- [ ] Any breaking changes?

### CODE REVIEW

File: sdk/storage/azure-storage-files-datalake/src/datalake_sas_builder.cpp [hunk 1]
```diff
@@ -223,8 +228,8 @@ namespace Azure { namespace Storage { namespace Sas {
         + "\n" + userDelegationKey.SignedService + "\n" + userDelegationKey.SignedVersion + "\n"
         + PreauthorizedAgentObjectId + "\n" + AgentObjectId + "\n" + CorrelationId + "\n"
         + (IPRange.HasValue() ? IPRange.Value() : "") + "\n" + protocol + "\n" + SasVersion + "\n"
-        + resource + "\n" + "\n" + CacheControl + "\n" + ContentDisposition + "\n" + ContentEncoding
-        + "\n" + ContentLanguage + "\n" + ContentType;
```
Comment (LarryOsterman) [inline sdk/storage/azure-storage-files-datalake/src/datalake_sas_builder.cpp:227]: Silly question: I don't seem to find anything in this code which digitally signs `stringToSign`. Would it be better named as `stringToEncode`?

Similarly, the variable `signature` doesn't appear to be a digital signature.

Comment (Jinming-Hu) [reply sdk/storage/azure-storage-files-datalake/src/datalake_sas_builder.cpp:227]: I think HMAC-256 can be considered a signing process.

**Review by Jinming-Hu — APPROVED** [review]

**Pull request merged**